snetd/cmd/serve.go

```diff
@@ -173,8 +172,8 @@ func (d daemon) start() {
 	}
 
 	if config.GetString(config.DaemonTypeKey) == "grpc" {
-		// set the maximum that the server can receive to 4GB. It is set to for 4GB because of issue here https://github.com/grpc/grpc-go/issues/1590
-		maxsizeOpt := grpc.MaxRecvMsgSize(math.MaxInt32)
+
+		maxsizeOpt := grpc.MaxRecvMsgSize(config.GetInt(config.MaxMessageSizeInMB) * 1024 * 1024)
```

Comment:
This will overflow on 32 bit platforms like raspberry pi, whose maximum int value is value is 2,147,483,647

I think we should limit to this value. People should be using streaming before they get to the point of sending 2 GB messages, so it shouldn't be overly restricting to anyone.

Reply:
Enforced this check on the config , [validate](https://github.com/singnet/snet-daemon/pull/241/files#diff-3baf47c64847a8fb8aaa8cc2e088513bR190) function